### ISSUE TITLE
Unsupported gpu architecture 'compute_21' 

### ISSUE THREAD

**Issue by sasacluar** [vlfeat/matconvnet#143]
I am trying to enable compilation for the GPUs but I can't land on a stable configuration set that works.  My latest attempt ended on this error: Unsupported gpu architecture 'compute_21' .  Please see below my current configuration and the error message.  Can someone recommend a stable configuration similar to this one and a solution to the 'compute_21' problem?  Thank you!

Matlab 2015a
MSVC 2011
CUDA Toolkit v7.0

  CUDADevice with properties:

                      Name: 'NVS 5400M'
                     Index: 1
         ComputeCapability: '2.1'
            SupportsDouble: 1
             DriverVersion: 7
            ToolkitVersion: 6.5000
        MaxThreadsPerBlock: 1024
          MaxShmemPerBlock: 49152
        MaxThreadBlockSize: [1024 1024 64]
               MaxGridSize: [65535 65535 65535]
                 SIMDWidth: 32
               TotalMemory: 1.0737e+09
           AvailableMemory: 775811072
       MultiprocessorCount: 2
              ClockRateKHz: 950000
               ComputeMode: 'Default'
      GPUOverlapsTransfers: 1
    KernelExecutionTimeout: 1
          CanMapHostMemory: 1
           DeviceSupported: 1
            DeviceSelected: 1

Error Message:
>> vl_compilenn('enableGpu', true, 'cudaMethod','nvcc')
vl_compilenn:	CUDA: MEX config file: 'C:\Program Files\MATLAB\R2015a\toolbox\distcomp\gpu\extern\src\mex\win64\mex_CUDA_win64.xml'
nvcc fatal   : Unsupported gpu architecture 'compute_21' 
Error using vl_compilenn>nvcc_compile (line 445)
Command "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v7.0\bin\nvcc" -c
"C:\Users\rcasas\Desktop\matconvnet-1.0-beta11\matconvnet-1.0-beta11\matlab\src\bits\impl\copy_gpu.cu" -DNDEBUG -DENABLE_GPU
-D__SSSE3__ -I"C:\Program Files\MATLAB\R2015a\extern\include" -I"C:\Program
Files\MATLAB\R2015a\toolbox\distcomp\gpu\extern\include" -Xcompiler /MD -gencode=arch=compute_21,code=\"sm_21,compute_21\"  -o
"C:\Users\rcasas\Desktop\matconvnet-1.0-beta11\matconvnet-1.0-beta11\matlab\mex\.build\bits\impl\copy_gpu.obj" failed.

Error in vl_compilenn>(parfor body) (line 391)
      nvcc_compile(opts, srcs{i}, toobj(bld_dir,srcs{i}), flags.nvcc) ;

Error in vl_compilenn (line 387)
parfor i = 1:numel(horzcat(lib_src, mex_src))

**Comment by pzhu10** [comment]
Hi,
 I also met  this issue a few days ago, I thinkI figure out what is going on!

![cudamethod-nvc](https://cloud.githubusercontent.com/assets/11349256/7587084/1a3e708a-f8e6-11e4-908b-f8b43afd71ef.jpg)

![compute-30](https://cloud.githubusercontent.com/assets/11349256/7587090/230c1b0e-f8e6-11e4-9f02-23ecffddbbaa.jpg)

when I used  compiling command vl_compilenn('enableGpu', true, 'cudaMethod','nvcc') on windows OS, cuDNN will also be used ( the default setting ), but the cuDNN library requires a GPU of compute capability 3.0 or higher. so unsupported gpu architecture 'compute_21'  error threw out.

After I upgraded my GPU to GTX750 ( compute capability is 5.0 ), this error disappeared.

hope helpful for you.
